compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -300,9 +325,29 @@ namespace cppmicroservices {
       return GetConfiguration(pid);
     }
 
-    std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> ConfigurationAdminImpl::ListConfigurations(const std::string& /* filter */)
+    std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> ConfigurationAdminImpl::ListConfigurations(const std::string& filter)
     {
-      throw std::invalid_argument("Method not currently implemented");
+      std::vector<std::shared_ptr<cppmicroservices::service::cm::Configuration>> result;
+      {
+        std::lock_guard<std::mutex> lk {configurationsMutex};
+
+        if (filter.empty()) {
```

Comment:
Can the filter string be invalid? Should we test for that? The OSGI spec says that ListConfigurations is supposed to throw a InvalidSyntaxException if the filter string is invalid.

Reply:
Creating an LDAPFilter instance using "filter" string that is invalid throws InvalidSyntaxException.
It also throws when an empty string is passed in. That's why I check for an empty string first, to see if the user didn't pass in an argument or passed in an empty string (no filter).

Reply:
The LDAPFilter ctor throws `std::invalid_argument`. `std::invalid_argument` can be thrown instead of `InvalidSyntaxException`.